.github/workflows/unittest.yml

```diff
@@ -0,0 +1,28 @@
+name: Unit Tests
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+
+jobs:
+  unittest:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.11]
+    if: ${{ !contains(github.event.head_commit.message, '[ci skip]') }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      - name: Test with unittest
+        run: |
+          as_api="http://localhost:4567" as_un="admin" as_pw="admin" python -m unittest discover test
```

Comment:
Will we need to be running a local dev instance of ASpace to get this to work/test properly? If so, I'll need to set up one for future work.

Reply:
No, this is just for GitHub's runners to be able to run our tests in GitHub Actions.  Adding this to CI means we shouldn't need to run tests locally again, really.  Though, if we adopt the vcr suggestion proposed in #58 , there will be some setup required to capture those initial recordings which would likely mean we'd want to have something running locally.  

Typically, we'd be setting these env vars in our local dotenv files, but since those aren't part of the codebase Github doesn't have access to them and we need to explicitly set them.  There are other more sophisticated ways to do this, but this was the most expedient with no obvious (to me) downsides.